sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -223,7 +234,8 @@ public HttpMethod getHttpMethod() {
      */
     @Override
     public int[] getExpectedStatusCodes() {
-        return CoreUtils.clone(expectedStatusCodes);
+//        return expectedStatusCodes;
+        return expectedStatusCodes == null ? null : expectedStatusCodes.stream().toArray();
```

Comment:
This method appears to only be used in tests and in one case where we check if a status code is valid. Thoughts on removing this method since it is non-public API and making callers of this use the `boolean isExpectedResponseStatusCode(int statusCode)` method? This will be more performant as we can completely remove the `int[]` cloning.

Reply:
The only caller to this is `HttpResponseBodyDecoder.isErrorStatus`

Reply:
This is unfortunately public API, otherwise I would have done what you suggested. I rearranged the code internally so that the only things calling this code is test code, which is why I pay the price for int[] cloning here, rather than in the constructor.

Reply:
My mistake - SwaggerMethodParser is itself a non-public class. I will revise the code.